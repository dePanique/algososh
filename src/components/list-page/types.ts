import { ElementStates } from "../../types/element-states";

export interface IObject {
    index: number
    state: ElementStates
    value: number
}

export type TActivness = {
    status: boolean,
    value: number,
    index?: number
}

export interface IHashTable<T> {
    [name: string]: {
        'topRow': {
            element: ''
        },
        'middleRow': {
            value: T,
            index: number,
            state: ElementStates
        },
        'bottomRow': {
            element: ''
        }
    } 
}