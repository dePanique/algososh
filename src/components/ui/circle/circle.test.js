import { Circle } from './circle';
import renderer from 'react-test-renderer';
import { ElementStates } from '../../../types/element-states';

describe('Circle tests', () => {
    it('should render Circle without text', () => {
        const tree = renderer
        .create(
            <Circle />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render Circle without text', () => {
        const tree = renderer.create(
            <Circle letter='qwerty'/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render Circle with head prop', () => {
        const tree = renderer.create(
            <Circle head/>
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    });

    it('should render Circle with react-element in head', () => {
        const tree = renderer.create(
            <Circle head={<Circle />}/>
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    });

    it('should render Circle with tail', () => {
        const tree = renderer.create(
            <Circle tail />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render Circle with react-element in tail', () => {
        const tree = renderer.create(
            <Circle head={<Circle />} />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render Circle with index', () => {
        const tree = renderer.create(
            <Circle index={14} />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render Circle with prop isSmall===true', () => {
        const tree = renderer.create(
            <Circle isSmall />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render Circle default state', () => {
        const tree = renderer.create(
            <Circle state={ElementStates.Default} />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render Circle changing state', () => {
        const tree = renderer.create(
            <Circle state={ElementStates.Changing} />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render Circle modified state', () => {
        const tree = renderer.create(
            <Circle state={ElementStates.Modified} />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render Circle modified state', () => {
        const tree = renderer.create(
            <Circle state={ElementStates.Modified} />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
})