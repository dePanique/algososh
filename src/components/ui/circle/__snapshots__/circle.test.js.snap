// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle tests should render Circle changing state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle tests should render Circle default state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle tests should render Circle modified state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle tests should render Circle modified state 2`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle tests should render Circle with head prop 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle tests should render Circle with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    14
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle tests should render Circle with prop isSmall===true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle tests should render Circle with react-element in head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle tests should render Circle with react-element in tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle tests should render Circle with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle tests should render Circle without text 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle tests should render Circle without text 2`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      qwerty
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
